File: app/javascript/react/containers/EventBrite.js
import React from 'react'
import EventBriteTile from '../components/EventBriteTile'

class EventBrite extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
        events: []
    }
    this.handleClick = this.handleClick.bind(this)
  }
  componentWillMount() {
    fetch('https://www.eventbriteapi.com/v3/events/search/?location.address=boston&subcategories=2003%2C2001%2C2002&search_type=networking&token=JLHHNB4KRCZWR7WZPKAY')
    .then(response => response.json())
    .then(body => {
      this.setState({
        events: body.events
      })
    })
  }

  addEvent(payLoad) {

  }

  handleClick(event) {
    event.preventDefault()

  }
  render () {
    let event = this.state.events.map(event => {
      let logo;
      debugger
      if(event.logo != null) {
        logo = event.logo.original.url
      } else {logo = "N/A"}
      return(
        <EventBriteTile
          key={event.id}
          id={event.id}
          name={event.name.text}
          description={event.description.text}
          url={event.url}
          logo={logo}
          handleClick={this.handleClick}
        />
      )
    })
    return(
      <div>
        <ol>
          {event}
        </ol>
      </div>
    )
  }
}

export default EventBrite;

File: app/models/eventBrite.rb
require 'httparty'

class Eventbrite

  def self.scrape
    @event_data = HTTParty.get("https://www.eventbriteapi.com/v3/events/search/?location.address=boston&subcategories=2003%2C2001%2C2002&search_type=networking&token=W6RPJZTBWTTQMJUDOSCB")
    @event_data['events'].each do |event|
      time = Time.parse(event['start']['utc']).getutc.strftime("%I:%M")
      end_time = Time.parse(event['end']['utc']).getutc.strftime("%I:%M")
      eventData = {
        name: event['name']['text'],
        description: event['description']['text'],
        time: time,
        date: event['start']['utc'],
        url: event['url'],
        event_id: event['id']
      }
      Event.create!(eventData)
    end
  end
end
